core/src/main/java/de/danoeh/antennapod/core/feed/FeedMedia.java

```diff
@@ -579,4 +580,12 @@ private void checkEmbeddedPicture() {
             hasEmbeddedPicture = Boolean.FALSE;
         }
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (o instanceof RemoteMedia) {
```

Comment:
Ah, then please provide a comment.

"// TODO: still want default equals comparison for most instances of FeedMedia, but need something better for RemoteMedia"

Though, now that I think about it, that still makes me uncomfortable.  it looks like the base class (FeedComponent) bases equality only on the id field.  @mfietz, do we leave it as is or should we do something more normal for .equals()?